src/features/organizations/utils/generateTreeData.spec.ts

```diff
@@ -0,0 +1,219 @@
+    const organizations: ZetkinOrganization[] = [
+      mockOrganization({
+        id: 1,
+        parent: null,
+        title: 'Party A',
+      }),
+      mockOrganization({
+        id: 2,
+        parent: null,
+        title: 'Party B',
+      }),
+    ];
+
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: { id: 1, parent: null, title: 'Party A' },
+        role: 'admin',
+      }),
+      mockMembership({
+        organization: { id: 2, parent: null, title: 'Party B' },
+        role: 'admin',
+      }),
+    ];
+
+    const result = generateTreeData(organizations, memberships);
+
+    const expectedTreeData: TreeItemData[] = [
+      {
+        avatar_file: null,
+        children: [],
+        country: 'SE',
+        email: null,
+        id: 1,
+        is_active: false,
+        is_open: false,
+        is_public: true,
+        lang: null,
+        parent: null,
+        phone: null,
+        slug: 'slug',
+        title: 'Party A',
+      },
+      {
+        avatar_file: null,
+        children: [],
+        country: 'SE',
+        email: null,
+        id: 2,
+        is_active: false,
+        is_open: false,
+        is_public: true,
+        lang: null,
+        parent: null,
+        phone: null,
+        slug: 'slug',
+        title: 'Party B',
+      },
+    ];
+
+    expect(result).toEqual(expectedTreeData);
+  });
+
+  it('ignores organizations without membership', () => {
+    const organizations: ZetkinOrganization[] = [
+      mockOrganization({
+        id: 1,
+        parent: null,
+        title: 'Party A',
+      }),
+      mockOrganization({
+        id: 2,
+        parent: null,
+        title: 'Party B',
+      }),
+    ];
+
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: { id: 3, parent: null, title: 'Party C' },
+        role: 'organizer',
+      }),
+      mockMembership({
+        organization: { id: 4, parent: null, title: 'Party D' },
+        role: 'admin',
+      }),
+      mockMembership({
+        organization: { id: 5, parent: null, title: 'Party E' },
+        role: null,
+      }),
+    ];
+
+    const result = generateTreeData(organizations, memberships);
+
+    const expectedTreeData: TreeItemData[] = [];
+
+    expect(result).toEqual(expectedTreeData);
+  });
+
+  it('considers a child org a top org if the user doesnt have access to parent org', () => {
+    const organizations: ZetkinOrganization[] = [
+      mockOrganization({
+        id: 1,
+        parent: null,
+        title: 'Party A',
+      }),
+      mockOrganization({
+        id: 2,
+        parent: {
+          id: 1,
+          title: 'Party A',
+        },
+        title: 'Party B',
+      }),
+    ];
+
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: {
+          id: 2,
+          parent: {
+            id: 1,
+            title: 'Party A',
+          },
```

Comment:
This is not the right format. The `organization` property on a `ZetkinMembership` cannot have a `parent`. It should only contain the ID and the title.